### PR TITLE
Add balance leader score option for release 3.0

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve? 
- [x] cherry pick from #1743 
- [x] set default strategy to 'bySize'
- [x] cherry pick from #2075
### Check List 

Tests 

 - Unit test

### Release note 

Add balance leader score option, It will be allowed to count leader score according to size or count.

### CODE REVIEW

File: tools/pd-ctl/README.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ Usage:
     "hot-region-cache-hits-threshold": 3,
     "hot-region-schedule-limit": 2,
     "leader-schedule-limit": 4,
+    "leader-schedule-policy": "count",
```
Comment (rleungx) [inline tools/pd-ctl/README.md:115]: should change to "size"?

File: server/config.go [hunk 2]
```diff
@@ -488,6 +489,8 @@ type ScheduleConfig struct {
 	MaxStoreDownTime typeutil.Duration `toml:"max-store-down-time,omitempty" json:"max-store-down-time"`
 	// LeaderScheduleLimit is the max coexist leader schedules.
 	LeaderScheduleLimit uint64 `toml:"leader-schedule-limit,omitempty" json:"leader-schedule-limit"`
+	// LeaderSchedulePolicy is the option to balance leader, there are some policies supported: ["count", "size"], default: "size"
+	LeaderSchedulePolicy string `toml:"leader-schedule-policy,omitempty" json:"leader-schedule-policy,string"`
```
Comment (nolouch) [inline server/config.go:493]:
```suggestion
	LeaderSchedulePolicy string `toml:"leader-schedule-policy" json:"leader-schedule-policy"`
```

File: server/config.go [hunk 3]
```diff
@@ -698,6 +706,11 @@ func IsDefaultScheduler(typ string) bool {
 	return false
 }
 
+// GetLeaderSchedulePolicy is to get leader schedule policy
+func (c *ScheduleConfig) GetLeaderSchedulePolicy() core.SchedulePolicy {
```
Comment (rleungx) [inline server/config.go:710]: Why we need it?

**Pull request merged**